Cython/Shadow.pyi

```diff
@@ -100,3 +103,19 @@ class typedef(CythonType, Generic[_T]):
 #    def __init__(self) -> None: ...
 
 #def fused_type(*args: Tuple[_T]) -> Type[FusedType[_T]]: ...
+
+class _EmptyDecoratorAndManager(object):
+    def __call__(self, x: _T) -> _T: ...
+
+    def __enter__(self) -> None: ...
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        tb: Optional[TracebackType]
+    ) -> None: ...
+
+cclass = ccall = cfunc = _EmptyDecoratorAndManager()
+
+def declare(t: Optional[Callable[..., _T]] = None, value: Any = ..., **kwds: Any) -> Optional[_T]: ...
```

Comment:
We can have an overload with typed `kwargs`

Reply:
Yeah good point, I don't think it needs to be an overload though.

Reply:
> I don't think it needs to be an overload though.

Why not? Would you consider `a = cython.declare(cython.int, 5, t=cython.int)` a valid syntax? (it isn't) It's either one or the other

Reply:
if you look at the [implementation](https://github.com/cython/cython/blob/8af0271186cc642436306274564986888d5e64c8/Cython/Shadow.py#L193) you can come up with more overloads.

Reply:
No, I simply meant that you don't need to overload it to provide types, just incorporate it directly into the function. See the following
https://github.com/cython/cython/blob/7b24bb6ef816649a25b741c43074adc45c31cf96/Cython/Shadow.pyi#L123-L128

Reply:
We can get away with this, but it would be more convenient if the type checking did not miss the misuse.